tests/Integration/IntegrationTest.php

```diff
@@ -0,0 +1,178 @@
+    ): void {
+        system("rm -rf " . escapeshellarg($outputPath));
+        system("mkdir " . escapeshellarg($outputPath));
+
+        $metas = new Metas([]);
+        $logger = new class($outputPath) extends AbstractLogger {
+
+            private $outputPath;
+
+            public function __construct($outputPath) {
+                $this->outputPath = $outputPath;
+            }
+
+            public function log($level, $message, array $context = []): void
+            {
+                $message = $level . ':' . $message . PHP_EOL;
+                file_put_contents($this->outputPath . '/log' . $level . '.txt', $message, FILE_APPEND);
+            }
+        };
+
+        $commandbus = QuickStart::create(
+            [
+                ParseFileCommand::class => new ParseFileHandler(
+                    $logger,
+                    new class implements EventDispatcherInterface
+                    {
+                        public function dispatch(object $event)
+                        {
+                            return $event;
+                        }
+                    },
+                    \phpDocumentor\Guides\Setup\QuickStart::createRstParser()
+                )
+            ]
+        );
+
+        $parseDirectoryHandler = new ParseDirectoryHandler(
+            new FileCollector($metas),
+            $commandbus,
+        );
+        $sourceFileSystem = new Filesystem(new Local(
+            $inputPath
+        ));
+        $sourceFileSystem->addPlugin(new Finder());
+
+        $parseDirCommand = new ParseDirectoryCommand(
+            $sourceFileSystem,
+            '',
+            'rst'
+        );
+
+        $documents = $parseDirectoryHandler->handle($parseDirCommand);
+        $compliler = new Compiler([
+            new MetasPass($metas),
+            new TransformerPass(
+                new DocumentNodeTraverser(
+                    [
+                        new TocNodeTransformer($metas)
+                    ]
+                )
+            )
+        ]);
+        $documents = $compliler->run($documents);
+
+        $renderer = \phpDocumentor\Guides\Setup\QuickStart::createRenderer($metas);
+        $renderDocumentHandler = new RenderDocumentHandler($renderer);
+
+        foreach ($documents as $document) {
+            try {
+                $renderDocumentHandler->handle(
+                    new RenderDocumentCommand(
+                        $document,
+                        RenderContext::forDocument(
+                            $document,
+                            $sourceFileSystem,
+                            new Filesystem(new Local($outputPath)),
+                            '/',
+                            $metas,
+                            new UrlGenerator(),
+                            'html'
+                        )
+                    )
+                );
+            } catch (\Exception $e) {
+                echo "Error:" . $e->getMessage() . PHP_EOL;
```

Comment:
Should we catch exceptions here? It would make sense to make tests fail on those.

Reply:
You are right.